src/compiler/diagnosticMessages.json

```diff
@@ -6227,6 +6227,11 @@
         "category": "Message",
         "code": 6718
     },
+    "Ensure that each file can have declaration emit generated without type information": {

```

Comment:
Both of these are true. it does allow one file at a time declaration emit, but it also ensures that not types need to be created by the type checker. I feel the latter constraint is the stricter one since it forbids most inference even if the information is available in the file.

Reply:
I think what’s going on in `localInferenceResolver.ts` is “type information,” but I take your point that needing to annotate function return types is probably more visible to users than the lack of cross-file information.